src/vfs2.c

```diff
@@ -1710,6 +1710,9 @@ int vfs2_add_uncommitted(sqlite3_file *file,
 	struct file *xfile = (struct file *)file;
 	PRE(xfile->flags & SQLITE_OPEN_MAIN_DB);
 	struct entry *e = xfile->entry;
+	if (e->page_size == 0) {
+		e->page_size = page_size;
```

Comment:
Should this be an error instead because there has not been a `PRAGMA page_size` as you state in the commit message?

Reply:
There's a choice here between requiring all nodes to run `PRAGMA page_size` when they open a DB connection and agree on a page size for a database out of band, and letting the leader that creates the first entry for a database set the page size, which is then propagated to the other nodes via raft. The first strategy is a bit less flexible but it does make the VFS's life simpler in this case, and it's what dqlite implements currently, so probably we should stick to it.

Reply:
This is fixed in a follow-up commit.